orbbec_camera/src/ros_service.cpp

```diff
@@ -1064,22 +1064,48 @@ void OBCameraNode::setSYNCHostimeCallback(
 }
 
 void OBCameraNode::sendSoftwareTriggerCallback(
-    const std::shared_ptr<std_srvs::srv::SetBool::Request>& request,
-    std::shared_ptr<std_srvs::srv::SetBool::Response>& response) {
-  try {
-    if (request->data) {
-      device_->triggerCapture();
+    const std::shared_ptr<CameraTrigger::Request>& request,
+    std::shared_ptr<CameraTrigger::Response>& response) {
+  if (!software_trigger_enabled_) {
+    service_capture_started_ = true; 
+    try {
+      if (request->data) {
+        RCLCPP_INFO_STREAM(logger_, "------------ Triggering");
+        device_->triggerCapture();
+      }
+
+      RCLCPP_INFO_STREAM(logger_, "------------ Capturing images");
+      sleep(service_capture_time_out_ / 1000.0f);
+
+      //capture_mutex_.lock();
+      RCLCPP_INFO_STREAM(logger_, "------------ Captured " << number_of_rgb_frames_captured_ << " " << number_of_depth_frames_captured_);
+      //capture_mutex_.unlock();
+
+      if (number_of_rgb_frames_captured_ == frames_per_trigger_ && number_of_depth_frames_captured_ == frames_per_trigger_) {
+        response->success = true;
+	response->image = *(color_image_);
+	response->depth_image = *(depth_image_);
+      }
+      else {
+        response->success = false;
+        response->message = "Failed to capture images";
+      }
+    } catch (const ob::Error& e) {
+      response->message = e.getMessage();
+      response->success = false;
+    } catch (const std::exception& e) {
+      response->message = e.what();
+      response->success = false;
+    } catch (...) {
+      response->message = "unknown error";
+      response->success = false;
     }
-    response->success = true;
-  } catch (const ob::Error& e) {
-    response->message = e.getMessage();
-    response->success = false;
-  } catch (const std::exception& e) {
-    response->message = e.what();
-    response->success = false;
-  } catch (...) {
-    response->message = "unknown error";
-    response->success = false;
+    service_capture_started_ = false;
+    number_of_rgb_frames_captured_ = 0;
+    number_of_depth_frames_captured_ = 0;
```

Comment:
could help maintaining this vortex of variables in these files if this was a teeny function called

```suggestion
    reset_capture_service_variables();
```

Reply:
Done